ofrak_core/ofrak/cli/ofrak_cli.py

```diff
@@ -143,29 +165,25 @@ def run(self, ofrak_env: OFRAKEnvironment, args: Namespace):
             exclude_components_missing_dependencies=args.exclude_components_missing_dependencies,
         )
 
-        if args.backend is not None:
-            if args.backend.lower() == "binary-ninja":
-                import ofrak_capstone  # type: ignore
-                import ofrak_binary_ninja  # type: ignore
-
-                ofrak.discover(ofrak_capstone)
-                ofrak.discover(ofrak_binary_ninja)
-
-            elif args.backend.lower() == "ghidra":
-                import ofrak_ghidra  # type: ignore
+        ofrak_pkgs = set(args.imports)
+        ofrak_pkgs.update(BACKEND_PACKAGES[args.backend])
 
-                ofrak.discover(ofrak_ghidra)
+        if not any(pkgs and pkgs.issubset(ofrak_pkgs) for pkgs in BACKEND_PACKAGES.values()):
+            logging.warning("No disassembler backend specified, so no disassembly will be possible")
 
-            elif args.backend.lower() == "angr":
-                import ofrak_capstone  # type: ignore
-                import ofrak_angr  # type: ignore
+        for ofrak_pkg_name in ofrak_pkgs:
+            if os.path.sep in ofrak_pkg_name or ofrak_pkg_name.endswith(".py"):
```

Comment:
Is it worth checking instead for `os.path.exists` and then cleaning up the path after that? This would enable importing directories with an `__init__.py` as well as DLLs that end in `.dll`, `.so`, or `.dylib`. 

Might be able to use some combination of these two recipes to import from files without path manipulations:

- <https://docs.python.org/3/library/importlib.html#importing-a-source-file-directly>
- <https://docs.python.org/3/library/importlib.html#approximating-importlib-import-module>

Reply:
I tried out a few things and did clean this up a little bit, but it's not clear to me what the correct Python procedure is to import files as you've described - the most correct way is not to do that, and make them modules instead!

I can keep looking into it if you can define the use case a little more, but at the moment it easily supports loading a Python file so you can quickly include some new scratch OFRAK extension as you're developing it. For everything else, I think we should just recommend installing it as a Python module we can import normally (i.e. by the module name).

Reply:
I really like where it's at right now, and don't think it's worth blocking on this.

Reply:
I had to revert the changes to imports and went back to the jankier-feeling `sys.path.append` method. The import technique shown [here](https://docs.python.org/3/library/importlib.html#importing-a-source-file-directly) breaks synthol discovery: the contents of the imported file are considered "builtins" by inspect and so get filtered out by discovery. I believe this is because that way of importing defines the modules in the top-level `__main__`, and [this](https://stackoverflow.com/a/56177941/16690095) comment says that classes defined in `__main__` (e.g. defined in interpreter) are considered "builtins". I haven't found clear info in the Python docs to verify all that, but it could explain what's going on.

Anyway I found this by adding a unit test for this import feature to the CLI to get full function coverage, so another +1 to that 100% function coverage policy! 

The informal quick check I used to see if it worked was "working" because it contained a `MagicDescriptionIdentifier.register(...)` call, and the file was successfully identified. But in fact, nothing in that file was being "discovered" - it is __imported__ though, and so the class was correctly added as a tag by the `MagicDescriptionIdentifier` which sort of backdoored a scratch file class into OFRAK without it being discovered. On a closer at that scratch file, I found that the components in that file were indeed not discovered and would not be run.